src/app/pages/products-list/card/card.component.ts

```diff
@@ -1,8 +1,28 @@
-import {Component} from '@angular/core';
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {Product} from 'src/app/shared/products/product.interface';
 
 @Component({
     selector: 'app-card',
     templateUrl: './card.component.html',
     styleUrls: ['./card.component.css'],
 })
-export class CardComponent {}
+export class CardComponent {
+    @Input() product: Product | null = null;
+    @Output() readonly buyedProduct: EventEmitter<Product | null> = new EventEmitter();
+
+    getProductImageUrl(index: number): string | null {
+        return this.product?.images && this.product?.images?.length > index
+            ? this.product?.images[index]?.url
+            : null;
+    }
+
+    getRating(index: number): boolean {
+        return this.product?.rating ? this.product?.rating >= index : false;
```

Comment:
Тут тоже `?.` лишний в выражении тернарника после `?`
```suggestion
        return this.product?.rating ? this.product.rating >= index : false;
```

Reply:
исправила

Reply:
Поинт почему именно так относится к сужению типизации в тернарнике - https://github.com/Letto228/Angular-learnjs-150424/pull/47#discussion_r1589925422